pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -31,38 +31,141 @@ import (
 
 type QueueTracker struct {
 	queueName           string
+	parent              *QueueTracker
+	queuePath           string
 	resourceUsage       *resources.Resource
 	runningApplications map[string]bool
-	maxResourceUsage    *resources.Resource
+	maxResources        *resources.Resource
 	maxRunningApps      uint64
 	childQueueTrackers  map[string]*QueueTracker
 }
 
 func newRootQueueTracker() *QueueTracker {
-	return newQueueTracker(configs.RootQueue)
+	qt := newQueueTracker(configs.RootQueue, nil)
+	return qt
 }
 
-func newQueueTracker(queueName string) *QueueTracker {
-	log.Logger().Debug("Creating queue tracker object for queue",
-		zap.String("queue", queueName))
+func newQueueTracker(queueName string, parent *QueueTracker) *QueueTracker {
 	queueTracker := &QueueTracker{
 		queueName:           queueName,
+		parent:              parent,
 		resourceUsage:       resources.NewResource(),
 		runningApplications: make(map[string]bool),
+		maxResources:        resources.NewResource(),
+		maxRunningApps:      0,
 		childQueueTrackers:  make(map[string]*QueueTracker),
 	}
+	parentQueueName := ""
+	if parent != nil {
+		queueTracker.queuePath = queueTracker.parent.queuePath + configs.DOT + queueName
+		parentQueueName = parent.queueName
+	} else if queueName == configs.RootQueue {
+		queueTracker.queuePath = configs.RootQueue
+	}
+	log.Logger().Debug("Created queue tracker object for queue",
+		zap.String("queue", queueName),
+		zap.String("parent", parentQueueName),
+		zap.String("queue path ", queueTracker.queuePath))
 	return queueTracker
 }
 
-func (qt *QueueTracker) increaseTrackedResource(queuePath string, applicationID string, usage *resources.Resource) error {
+type trackingType int
+
+const (
+	none trackingType = iota
+	user
+	group
+)
+
+func (qt *QueueTracker) increaseTrackedResource(queuePath string, applicationID string, trackType trackingType, usage *resources.Resource) bool {
 	log.Logger().Debug("Increasing resource usage",
+		zap.Int("tracking type", int(trackType)),
 		zap.String("queue path", queuePath),
 		zap.String("application", applicationID),
 		zap.Stringer("resource", usage))
 	if queuePath == "" || applicationID == "" || usage == nil {
-		return fmt.Errorf("mandatory parameters are missing. queuepath: %s, application id: %s, resource usage: %s",
-			queuePath, applicationID, usage.String())
+		return false
+	}
+
+	finalResourceUsage := qt.resourceUsage.Clone()
+	finalResourceUsage.AddTo(usage)
+	wildCardQuotaExceeded := false
+
```

Comment:
is this a new app to add to the list of running applications?
```
existingApp := runningApplications[appId]
```
Only check maxRunningApps within range if `existingApp == false`